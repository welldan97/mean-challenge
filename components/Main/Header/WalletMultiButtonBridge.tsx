import { WalletMultiButton } from '@solana/wallet-adapter-react-ui';

export default WalletMultiButton;
