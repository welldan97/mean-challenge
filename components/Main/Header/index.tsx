import { memo } from 'react';
import dynamic from 'next/dynamic';

// use dynamic import to mock in tests
const WalletMultiButton = dynamic(() => import('./WalletMultiButtonBridge'), {
  ssr: false,
});

const Header = memo(() => (
  <header className="sticky justify-end items-center top-0 z-30 flex w-full text-base-content shadow-sm bg-base-300">
    <h1 className="flex-1 p-4 text-l font-bold">Solana Balance</h1>
    <div className="mr-16">
      <WalletMultiButton className="btn" />
    </div>
  </header>
));

export default Header;
