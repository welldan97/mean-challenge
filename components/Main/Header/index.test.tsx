// import { WalletMultiButton } from '@solana/wallet-adapter-react-ui';
import { render } from '@testing-library/react';

import Header from '.';

jest.mock('next/dynamic', () => () => () => <span>WalletMultiButton</span>);

describe('Header', () => {
  test('render', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
});
