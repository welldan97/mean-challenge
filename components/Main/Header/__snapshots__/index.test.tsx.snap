// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render 1`] = `
<div>
  <header
    class="sticky justify-end items-center top-0 z-30 flex w-full text-base-content shadow-sm bg-base-300"
  >
    <h1
      class="flex-1 p-4 text-l font-bold"
    >
      Solana Balance
    </h1>
    <div
      class="mr-16"
    >
      <span>
        WalletMultiButton
      </span>
    </div>
  </header>
</div>
`;
